app/models/samples_report.rb

```diff
@@ -68,6 +68,11 @@ def calculate_lod_and_lob
     self.save
   end
 
+  def target_batch
+    # The target batch for this box is the batch of any sample which distractor is false or null
+    samples_report_samples.joins(:sample).where("samples.distractor IS NULL OR samples.distractor = false").first.sample.batch
```

Comment:
FYI: this is acceptable as we don't call the method much, but this will make a number of SQL requests:

```sql
select * from samples_reports_samples limit 1
select * from samples joins limit 1
select * from batches limit 1
```

The ideal would be to select from batch and use joins to find it from samples reports. I just tried, and the following should work...

```ruby
Batch
  .joins(:samples => :samples_report_sample)
  .where("samples_report_samples.samples_report_id = ?", id)
  .where("samples.distractor <> ?", true)
  .take
```

... if you must add the missing association on `Sample`:

```ruby
class Sample
  has_one :samples_report_sample
end

This is a problem of the active-record pattern, it's easy to use the object relations, and we can easily lose track of how many requests it will actually make in the background.

Reply:
Also you should memoize the object, for example `@target_batch ||= ...`, otherwise each call to the method will run the SQL request(s).